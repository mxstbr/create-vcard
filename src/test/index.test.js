// @flow
import vCardJS from 'vcards-js';
import createVCard from '../';

// The REV in a vCard string is the current date
// let's remove it since it breaks tests
const removeRev = (vcard: string) => {
  return vcard.replace(/\n?^REV:.*$/m, '');
};

it('should create a vcard-js', () => {
  expect(createVCard({})).toMatchSnapshot();
});

it('should attach simple properties', () => {
  expect(
    createVCard({
      firstName: 'Max',
      lastName: 'Stoiber',
    })
  ).toMatchSnapshot();
});

it('should correct attach photo URLs', () => {
  expect(
    createVCard({
      photo: 'https://mxstbr.com/headshot.jpeg',
    })
  ).toMatchSnapshot();
});

it('should convert the vCard to a string', () => {
  const string = createVCard({
    firstName: 'Max',
    lastName: 'Stoiber',
  }).getFormattedString();
  expect(removeRev(string)).toMatchSnapshot();
});
