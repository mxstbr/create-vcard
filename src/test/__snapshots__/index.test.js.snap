// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should attach simple properties 1`] = `
Object {
  "birthday": "",
  "cellPhone": "",
  "email": "",
  "firstName": "Max",
  "formattedName": "",
  "gender": "",
  "getFormattedString": [Function],
  "getMajorVersion": [Function],
  "homeAddress": Object {
    "city": "",
    "countryRegion": "",
    "label": "",
    "postalCode": "",
    "stateProvince": "",
    "street": "",
  },
  "homeFax": "",
  "homePhone": "",
  "lastName": "Stoiber",
  "logo": Object {
    "attachFromUrl": [Function],
    "base64": false,
    "embedFromFile": [Function],
    "embedFromString": [Function],
    "mediaType": "",
    "url": "",
  },
  "middleName": "",
  "namePrefix": "",
  "nameSuffix": "",
  "nickname": "",
  "note": "",
  "organization": "",
  "pagerPhone": "",
  "photo": Object {
    "attachFromUrl": [Function],
    "base64": false,
    "embedFromFile": [Function],
    "embedFromString": [Function],
    "mediaType": "",
    "url": "",
  },
  "role": "",
  "saveToFile": [Function],
  "socialUrls": Object {
    "facebook": "",
    "flickr": "",
    "linkedIn": "",
    "twitter": "",
  },
  "source": "",
  "title": "",
  "uid": "",
  "url": "",
  "version": "3.0",
  "workAddress": Object {
    "city": "",
    "countryRegion": "",
    "label": "",
    "postalCode": "",
    "stateProvince": "",
    "street": "",
  },
  "workEmail": "",
  "workFax": "",
  "workPhone": "",
  "workUrl": "",
}
`;

exports[`should convert the vCard to a string 1`] = `
"BEGIN:VCARD
VERSION:3.0
FN;CHARSET=UTF-8:Max Stoiber
N;CHARSET=UTF-8:Stoiber;Max;;;

END:VCARD
"
`;

exports[`should correct attach photo URLs 1`] = `
Object {
  "birthday": "",
  "cellPhone": "",
  "email": "",
  "firstName": "",
  "formattedName": "",
  "gender": "",
  "getFormattedString": [Function],
  "getMajorVersion": [Function],
  "homeAddress": Object {
    "city": "",
    "countryRegion": "",
    "label": "",
    "postalCode": "",
    "stateProvince": "",
    "street": "",
  },
  "homeFax": "",
  "homePhone": "",
  "lastName": "",
  "logo": Object {
    "attachFromUrl": [Function],
    "base64": false,
    "embedFromFile": [Function],
    "embedFromString": [Function],
    "mediaType": "",
    "url": "",
  },
  "middleName": "",
  "namePrefix": "",
  "nameSuffix": "",
  "nickname": "",
  "note": "",
  "organization": "",
  "pagerPhone": "",
  "photo": Object {
    "attachFromUrl": [Function],
    "base64": false,
    "embedFromFile": [Function],
    "embedFromString": [Function],
    "mediaType": undefined,
    "url": "https://mxstbr.com/headshot.jpeg",
  },
  "role": "",
  "saveToFile": [Function],
  "socialUrls": Object {
    "facebook": "",
    "flickr": "",
    "linkedIn": "",
    "twitter": "",
  },
  "source": "",
  "title": "",
  "uid": "",
  "url": "",
  "version": "3.0",
  "workAddress": Object {
    "city": "",
    "countryRegion": "",
    "label": "",
    "postalCode": "",
    "stateProvince": "",
    "street": "",
  },
  "workEmail": "",
  "workFax": "",
  "workPhone": "",
  "workUrl": "",
}
`;

exports[`should create a vcard-js 1`] = `
Object {
  "birthday": "",
  "cellPhone": "",
  "email": "",
  "firstName": "",
  "formattedName": "",
  "gender": "",
  "getFormattedString": [Function],
  "getMajorVersion": [Function],
  "homeAddress": Object {
    "city": "",
    "countryRegion": "",
    "label": "",
    "postalCode": "",
    "stateProvince": "",
    "street": "",
  },
  "homeFax": "",
  "homePhone": "",
  "lastName": "",
  "logo": Object {
    "attachFromUrl": [Function],
    "base64": false,
    "embedFromFile": [Function],
    "embedFromString": [Function],
    "mediaType": "",
    "url": "",
  },
  "middleName": "",
  "namePrefix": "",
  "nameSuffix": "",
  "nickname": "",
  "note": "",
  "organization": "",
  "pagerPhone": "",
  "photo": Object {
    "attachFromUrl": [Function],
    "base64": false,
    "embedFromFile": [Function],
    "embedFromString": [Function],
    "mediaType": "",
    "url": "",
  },
  "role": "",
  "saveToFile": [Function],
  "socialUrls": Object {
    "facebook": "",
    "flickr": "",
    "linkedIn": "",
    "twitter": "",
  },
  "source": "",
  "title": "",
  "uid": "",
  "url": "",
  "version": "3.0",
  "workAddress": Object {
    "city": "",
    "countryRegion": "",
    "label": "",
    "postalCode": "",
    "stateProvince": "",
    "street": "",
  },
  "workEmail": "",
  "workFax": "",
  "workPhone": "",
  "workUrl": "",
}
`;
